README.md

```diff
@@ -1,126 +1,241 @@
 
-#### Manual (non "pinned") Build Instructions
+For the curious, the "pinned" terminology comes from our days developing on macOS where you would run `brew pin` to lock one of the Leap dependencies to a specific version despite available upgrades.
 
-<details>
-  <summary>Ubuntu 20.04 & 22.04 Build Instructions</summary>
+#### Pinned Build
+From a terminal in the root of the `leap` repo, run our script to install dependencies.
+```bash
+sudo scripts/install_deps.sh
+```
+- If you are in an Ubuntu docker container, omit `sudo` because you run as `root` by default.
 
-Install required dependencies: 
+Next, call our pinned build script. You have to give it three arguments, a folder for dependencies, a build folder, and [the number of jobs or CPU cores/threads to use](#jobs-flag). This command uses a `deps` and `build` folder in the root of the Leap repo for the first two arguments, then builds using all but two of your computer's CPU threads.
+```bash
+scripts/pinned_build.sh deps build "$(( $(nproc) - 2 ))"
 ```
-apt-get update && apt-get install   \
-        build-essential             \
-        cmake                       \
-        curl                        \
-        git                         \
-        libboost-all-dev            \
-        libcurl4-openssl-dev        \
-        libgmp-dev                  \
-        libssl-dev                  \
-        libusb-1.0-0-dev            \
-        llvm-11-dev                 \
+- You don't need `sudo` for this command.
+
+The binaries will be in the Leap build directory you passed to the build script. For example, the command above leaves them in `build/bin` from the root of the Leap repository.
+
+#### Unpinned Build
+These instructions are valid for this branch. Other release branches may have different requirements so ensure you follow the directions in the branch or release you intend to build.
+
+<details> <summary>Ubuntu 22.04 Jammy & Ubuntu 20.04 Focal</summary>
+
+Install dependencies.
+```bash
+sudo apt-get update
+sudo apt-get install -y \
+        build-essential \
+        cmake \
+        curl \
+        git \
+        libboost-all-dev \
+        libcurl4-openssl-dev \
+        libgmp-dev \
+        libssl-dev \
+        libusb-1.0-0-dev \
+        llvm-11-dev \
         pkg-config
 ```
-and perform the build:
-```
-git submodule update --init --recursive
-mkdir build
+- If you are in an Ubuntu docker container, omit `sudo` because you run as `root` by default.
+From a terminal in the root of the `leap` repo, build.
+```bash
+mkdir -p build
 cd build
 cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=/usr/lib/llvm-11 ..
-make -j $(nproc) package
+make -j "$(( $(nproc) - 2 ))" package
```

Comment:
Is this a bashism? I don't recall how widely implemented `$(())` is. What happens if user only has two cores? Change seems unnecessary imo

Reply:
Good question. I don't know, so I tried it in DASH (used as `/bin/sh` in Ubuntu and other Debian-family Linuxes).

![2022-10-17 19-08-19 - arithmetic in DASH](https://user-images.githubusercontent.com/34947245/196300291-971dcdde-23ac-458a-975e-e490f73f4a84.png)

It worked! I don't use ZSH and I don't know if we're expected to support it, but I found online [z-shell also supports arithmetic operations](https://www.csse.uwa.edu.au/programming/linux/zsh-doc/zsh_10.html).

I also found online they still sell dual-core processors. I did not know that. I thought they stopped selling dual cores ten years ago. That's my bad. Even if that were true, you could have a dual-core virtual machine. I tried passing `0` and `-1` to `make -j` in a virtual machine and it rejected it.

It is also just obtuse syntax if we are writing this with an eye towards less experienced customers. I will remove it.

Reply:
Removed.